### ISSUE TITLE
Storybook addons and type definitions should be in `devDependencies`

### ISSUE THREAD

**Issue by matthieu-foucault** [thebuttonclan/pangolin#337]
**Specifications**

`@button-inc/bcgov-theme@1.0.1`

**Describe the bug**

npm reports warning with missing peer dependencies. However, those are development tools and should not be required when using the component library.

```
warning "@button-inc/bcgov-theme > @button-inc/storybook-progressive-enhancement-addon@0.0.9" has unmet peer dependency "@storybook/addons@^6.1.14".
warning "@button-inc/bcgov-theme > @button-inc/storybook-progressive-enhancement-addon@0.0.9" has unmet peer dependency "@storybook/api@^6.1.14".
warning "@button-inc/bcgov-theme > @button-inc/storybook-progressive-enhancement-addon@0.0.9" has unmet peer dependency "@storybook/components@^6.1.14".
warning "@button-inc/bcgov-theme > @button-inc/storybook-progressive-enhancement-addon@0.0.9" has unmet peer dependency "@storybook/core-events@^6.1.14".
warning "@button-inc/bcgov-theme > @button-inc/storybook-progressive-enhancement-addon@0.0.9" has unmet peer dependency "@storybook/theming@^6.1.14".
```

Looking at the `bcgov-theme`'s `package.json`, it looks like the following dependencies should be moved to `devDependencies`:

```
"@button-inc/storybook-progressive-enhancement-addon": "^0.0.9",
"@storybook/addon-links": "^6.2.9",
"@types/lodash": "^4.14.168",
"@types/styled-components": "^5.1.4"
```

**Comment by kriscooke** [comment]
Agreed - I've noticed this as well. Thanks!

**Comment by matthieu-foucault** [comment]
@kriscooke there is a vulnerability in one of the dependencies of `@button-inc/storybook-progressive-enhancement-addon` so this fix is getting a bit more important. As we use `improved-yarn-audit` we can easily ignore the CVE advisory though.

┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular Expression Denial of Service in prismjs              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ prismjs                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=1.25.0                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @button-inc/bcgov-theme                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @button-inc/bcgov-theme >                                    │
│               │ @button-inc/storybook-progressive-enhancement-addon >        │
│               │ react-syntax-highlighter > refractor > prismjs               │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1002422                     │
└───────────────┴──────────────────────────────────────────────────────────────┘

**Comment by pbastia** [comment]
Thanks @kriscooke! Our team is going to start publishing a couple of npm packages that depend on this toolkit, do you know when we'll be able to remove these packages from the dependency list?
